code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt

```diff
@@ -0,0 +1,240 @@
+    }
+
+    @Test
+    fun `construct should map valid data to notification data and view for Intent Data`() {
+        val pushTemplate = BasicPushTemplate(IntentData(mockBundle, null))
+        val notification = AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            CHANNEL_ID_TO_USE,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            CONTAINER_LAYOUT_VIEW_ID
+        ).build()
+
+        verifyNotificationDataFields(notification, pushTemplate)
+        verifyNotificationViewDataAndColors(pushTemplate)
+    }
+
+    @Test
+    fun `construct should not set notification sound if pushTemplate sound is invalid`() {
+        dataMap.replaceValueInMap(
+            Pair(
+                PushTemplateConstants.PushPayloadKeys.SOUND,
+                "invalid_sound"
+            )
+        )
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+        val notification =
+            AEPPushNotificationBuilder.construct(
+                context,
+                pushTemplate,
+                CHANNEL_ID_TO_USE,
+                trackerActivityClass,
+                smallLayout,
+                expandedLayout,
+                CONTAINER_LAYOUT_VIEW_ID
+            ).build()
+
+        assertNull(notification.sound)
+    }
+
+    @Test
+    fun `test createIntent returns intent with mapped properties`() {
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+        val action = TIMER_EXPIRED
+        val intent = AEPPushNotificationBuilder.createIntent(action, pushTemplate)
+
+        assertEquals(action, intent.action)
+        assertEquals(Intent.FLAG_ACTIVITY_SINGLE_TOP, intent.flags)
+        assertNotNull(intent.extras)
+    }
+
+    @Config(sdk = [25])
+    @Test
+    fun `construct should set priority and vibration for API level below 26`() {
+        val pushTemplate = BasicPushTemplate(MapData(dataMap))
+        val notification =
+            AEPPushNotificationBuilder.construct(
+                context,
+                pushTemplate,
+                CHANNEL_ID_TO_USE,
+                trackerActivityClass,
+                smallLayout,
+                expandedLayout,
+                CONTAINER_LAYOUT_VIEW_ID
+            ).build()
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            assertEquals(NotificationCompat.PRIORITY_HIGH, notification.priority)
+            assertArrayEquals(LongArray(0), notification.vibrate)
+        }
+    }
+
+    private fun verifyNotificationDataFields(
+        notification: Notification,
+        pushTemplate: AEPPushTemplate
+    ) {
+        val pendingIntent = notification.contentIntent
+        val shadowPendingIntent = Shadows.shadowOf(pendingIntent)
+        val intent = shadowPendingIntent.savedIntent
+
+        assertEquals(Notification::class.java, notification.javaClass)
+        assertEquals(pushTemplate.ticker, notification.tickerText)
```

Comment:
nit: can we also verify the channel id, badge count, visibility, sound and priority settings here, if possible?

Reply:
sure will add them